dev/com.ibm.ws.logging.core/src/com/ibm/websphere/logging/hpel/LogRecordContext.java

```diff
@@ -298,41 +281,29 @@ public static void getExtensions(Map<String, String> map)
             return;
         }
         recursion.set(Boolean.TRUE);
-        LinkedList<String> cleanup = new LinkedList<String>();
-        r.lock();
+        HashMap<String, WeakReference<Extension>> cleanup = new HashMap<>();
```

Comment:
Why not use a synchronized class here instead since you are removing the locking?

Reply:
The previous locking allowed shared locks for read operations and exclusive locks for write operations.  The same is done with ConcurrentHashMap under the covers.  We do not want to use a completely synchronized Map because that would become a bottleneck since read operations are done for every single logging and tracing invocation.